packages/data-stores/src/site/actions.ts

```diff
@@ -444,7 +445,9 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		}
 
 		const response: { blog: string } = yield wpcomRequest( {
-			path: `/sites/${ encodeURIComponent( siteSlug ) }/theme-setup`,
+			path: `/sites/${ encodeURIComponent( siteSlug ) }/theme-setup${
+				locale ? `/?_locale=${ locale }` : ''
```

Comment:
I guess we can simply pass `_locale=user` to use the user locale for the headstart. Also, we have to replace `get_locale` with `determine_locale` in the theme-setup endpoint to support this way. WDYT?

See fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Serfg%2Qncv%2Qcyhtvaf%2Sraqcbvagf%2Sfvgrf%2Qgurzr%2Qfrghc.cuc%3Se%3Q6sn59r0o%23136-og

Reply:
@markbiek, WordPress core includes a mechanism for `_locale=user` to be used as a simpler, non-dynamic way to request that the server locale be set to the current user's locale. That code is in the `determine_locale()` function, as illustrated by the lines below in WordPress 6.2.

https://github.com/WordPress/wordpress-develop/blob/bea4307daa21a3f97d159898b0ac676332e0e7de/src/wp-includes/l10n.php#L145-L147

Reply:
@daledupreez Are you referring to the comment about passing the user locale for the headstart? Or the change in this PR?

Because, as far as I can tell, passing the `_locale` param is working as the PR stands now.

Reply:
Both approaches will work. However, always passing `?_locale=user` in the URL will result in simpler client-side code, because we don't need to know what the user locale is, and we don't need to pass it around in the code. And I believe that is mostly what @arthur791004 was suggesting as a minor improvement/simplification.

Reply:
Ohhhh, I see. You mean the literal string `_locale=user` as opposed to `_locale=<the user's locale>`.

That makes total sense and I'll get that updated.

Thanks for clarifying.

Reply:
> You mean the literal string `_locale=user` as opposed to `_locale=<the user's locale>`

Yeah, this is what I mean. Sorry for the misunderstanding 😅

I'll make a change on this endpoint to make the mechanism for `_locale=user` work